src/main/kotlin/com/openlattice/data/integration/destinations/BaseS3Destination.kt

```diff
@@ -88,17 +88,17 @@ abstract class BaseS3Destination(
             val entities = data.map { Entity(it.key, it.details) }
 
             val edges = data.map {
-                val srcDataKey = EntityDataKey(it.src.entitySetId, entityKeyIds[it.src])
-                val dstDataKey = EntityDataKey(it.dst.entitySetId, entityKeyIds[it.dst])
-                val edgeDataKey = EntityDataKey(it.key.entitySetId, entityKeyIds[it.key])
+                val srcDataKey = EntityDataKey(it.src.entitySetId, entityKeyIds.getValue(it.src))
+                val dstDataKey = EntityDataKey(it.dst.entitySetId, entityKeyIds.getValue(it.dst))
+                val edgeDataKey = EntityDataKey(it.key.entitySetId, entityKeyIds.getValue(it.key))
                 DataEdgeKey(srcDataKey, dstDataKey, edgeDataKey)
             }.toSet()
             integrateEntities(entities, entityKeyIds, updateTypes) + createAssociations(edges)
         }
     }
 
     private fun uploadToS3WithRetry(s3entitiesAndValues: List<Pair<S3EntityData, ByteArray>>) {
-        var currentRetryCount = 0
+        val currentRetryCount = 0
```

Comment:
this seems backwards to me -- we should be incrementing this counter somewhere, not making it immutable. :)

Reply:
probably just pull this out to a separate issue issue. Otherwise changes lgtm.

Reply:
Probably yes, I just fixed some warnings in the code without actually looking at the implementation. @geekbeast What was this `currentRetryCount` supposed to do?